cmd/otelcol/main.go

```diff
@@ -41,7 +42,9 @@ func main() {
 		GitHash:  version.GitHash,
 	}
 
-	svc, err := service.New(service.Parameters{ApplicationStartInfo: info, Factories: factories})
+	rp := resourcedetection.NewResourceProvider()
```

Comment:
For example, to use AWS & GCP custom detectors, a user would add the following:

```go
rp.AddDetectors(awsdetector.EC2, gcpdetector.GCE, gcpdetector.GKE)
```

Reply:
Would there be a set of default detectors?

Reply:
The `FromEnv` detector is added by default here: https://github.com/open-telemetry/opentelemetry-collector/pull/1059/files#diff-1126dd443ec86da518db395b09a6ca12R37